flocker/acceptance/endtoend/test_dockerplugin.py

```diff
@@ -20,6 +25,36 @@ class DockerPluginTests(TestCase):
     """
     Tests for the Docker plugin.
     """
+    def restart_docker(self, address):
+        """
+        Restart the Docker daemon on the specified address.
+
+        :param bytes address: The public IP of the node on which Docker will
+            be restarted.
+        """
+        distro = []
+        get_distro = run_ssh(
+            reactor, b"root", address, ["python", "-m", "platform"],
+            handle_stdout=distro.append)
+        get_distro.addCallback(lambda _: distro[0].lower())
+
+        def restart_docker(distribution):
+            if 'ubuntu' in distribution:
+                command = ["service", "docker", "restart"]
+            else:
+                command = ["systemctl", "restart", "docker"]
```

Comment:
Although the `service` command seems to be available in Centos-7 where it redirects to systemctl.
```
[root@centos-flocker-node1 ~]# service docker restart
Restarting docker (via systemctl):                         [  OK  ]
```

I'd be inclined to use that instead of detecting the distro.